tool/infocert.c

```diff
@@ -0,0 +1,119 @@
+/******************************************************************************
+ *
+ * Copyright 2018 Xaptum, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+#include "infocert.h"
+#include <xtt/crypto_types.h>
+#include <xtt/certificates.h>
+#include <xtt/util/asn1.h>
+#include <xtt/util/root.h>
+#include <xtt/util/util_errors.h>
+#include <xtt/util/file_io.h>
+#include <string.h>
+#include <stdio.h>
+
+struct xtt_server_certificate_raw_type;
+
+static
+void server_cert_out(unsigned char* certificate){
+    unsigned char* server_id = xtt_server_certificate_access_id((struct xtt_server_certificate_raw_type*) certificate);
```

Comment:
Rather than doing the cast to `xtt_server_certificate_raw_type` inside of each `access` call, let's do that once:

I suggest changing `server_cert_out(unsigned char*)` to `server_cert_out(struct xtt_server_certificate_raw_type*)`. Then do the cast in the `info_cert()` function.

Or, leave the `server_cert_out(unsigned char*)` signature as is, but do the cast once in the first line:
```
void server_cert_out(unsigned char* raw) {
    struct xtt_server_certificate_raw_type* cert = (struct xtt_server_certificate_raw_type*) raw;
    printf("Server Id: ");
    ...
}
```